Polyfills/Canvas/Source/Colors.h

```diff
@@ -162,40 +162,52 @@ namespace Babylon::Polyfills::Internal
             {"yellow", 0xffff00},
             {"yellowgreen", 0x9acd32} };
 
-        if (str[0] == '#' && str.length() == 4)
+        if (str[0] == '#')
```

Comment:
Why don't we use regex in this case too? Also will you add a comment with example strings we are trying to match (like the later section does for `rgb(#,#,#)` or `rgba(#,#,#,#)`)? Having the example strings makes it much easier to understand what the code is trying to do.

Reply:
I thought the same thing, but I'm not sure regex is much better for this case. I think having unit tests will cover the example strings.

Reply:
I'm not a fan of regex. I did a quick test to parse 1 million rgba color strings. Timings in release:
Time for RegExp : 17.5821 seconds
Time for scanf : 1.2384 seconds
Time for strtol : 0.1339 seconds
We should discuss this sometimes.

Reply:
Does the RegExp time include the time to parse the initial regex string?

Reply:
No, std::regex is instanciated once. If it's instanciated for each color parsing, I get these values:

Time for RegExp all: 33.9778 seconds
Time for scanf all: 1.2475 seconds
Time for strtol all: 0.1473 seconds

Which is less than I anticipated. Basically, computing the regex object is more of less the same than parsing the color.